packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -989,6 +989,9 @@ class FlutterVmService {
         }
       }
       return await extensionAdded.future;
+    } on vm_service.RPCError {
+      // Translate this exception into something the outer layer understands
+      throw VmServiceDisappearedException();
```

Comment:
Actually, as I recall, RPCError has some information, like a code? Should we preserve that information? Check for it?